packages/contracts-bedrock/tasks/update-registry.ts

```diff
@@ -0,0 +1,62 @@
+import fs from 'fs'
+import path from 'path'
+
+import { task } from 'hardhat/config'
+
+task('update-registry', 'Update OP Stack Chain Registry')
+  .addParam('registry', 'The address of the OpStackChainRegistry contract')
+  .addParam(
+    'deploymentFolder',
+    'The path to the folder containing the deployment files'
+  )
+  .setAction(async (args, hre) => {
+    const OpStackChainRegistry = await hre.ethers.getContractFactory(
+      'OpStackChainRegistry'
+    )
+    const registry = OpStackChainRegistry.attach(args.registry)
+
+    const deploymentFolder = path.resolve(__dirname, args.deploymentFolder)
+
+    const deploymentFiles = fs
```

Comment:
There is a better way to get the deployments using hardhat deploy, `hre.deployments.get`
https://github.com/wighawag/hardhat-deploy#the-deployments-field
be sure to import hh deploy into the file to get the ts lsp to work

Reply:
Doesn't this just get the deployment artifact for a single contract? Wouldn't you still need a list of names? Or do you see that as something you'd provide as an arg?

The way I have it now is it asks if the deployment (eg 'mainnet') is already claimed, if not it claims it. Then it goes through each of the files, asks if it's already registered, if not register it. 

How did you envision this?